pkg/cluster/backupstorage/storage.go

```diff
@@ -0,0 +1,17 @@
+package backupstorage
+
+import "k8s.io/kubernetes/pkg/api"
+
+// Storage defines the underlying storage used by backup sidecar.
+type Storage interface {
+	// Init will initialize an empty storage.
+	// It takes place on non-restore path.
+	Init() error
+	// Clone will try to clone another storage referenced by cluster name.
+	// It takes place on restore path.
+	Clone(from string) error
+	// Delete will delete this storage.
+	Delete() error
+	// AttachTo attaches this storage to PodSpec.
+	AttachToPodSpec(*api.PodSpec) *api.PodSpec
```

Comment:
or maybe we call this ConfigPod(*api.PodSpec) // config pod to use the storage.

not sure. do not feel very good about this interface.